### ISSUE TITLE
digicam_baseline vs baseline_from_random_triggers

### ISSUE THREAD

**Issue by dneise** [cta-sst-1m/digicampipe#146]
After the interesting discussion about the baseline in context with the "R1 & DL0 Telescope Event Interfaces and Prototype Evaluation" document, I wanted to execute the request from @Etienne12345 s email which said:
```
TODO
    * Someone: estimate degradation of sensitivity induced by one baseline every 2 seconds rather than for every event
```

So .... I went ahead and wanted to see the impact on the hillas parameters in the crab data sample from last november. 
So I executed `pipeline_crab.py` on the data sample and then went on to modify the pipeline in such a way that the baseline is only updated once every 2 seconds. Only then I realized that the pipeline *is not using the digicam baseline at all*. 

Is this because `pipeline_crab.py` is not our "std analysis" but just some possible example?

If so, do we have a std-analysis? And if so ... why is it not implemented in pipeline_crab.py? Wouldn't that be a good place to document what we think at the moment is the best way to analyse our digicam data?

**Comment by Etienne12345** [comment]
Indeed, afaik we use the baseline calculated from the interleaved triggers. I guess that we do so because the per-event baseline wasn't available since the very beginning, and nobody took the time to add it to digicampipe yet. I guess that if you feel like it you can go ahead and add it (tbc with Cyril of course).

**Comment by calispac** [comment]
@dneise you can go ahead with what you propose. This is more evaluating the impact on the full reconstruction. 

I was thinking of doing it using Monte Carlo. We would evaluate the baseline precision as function of NSB rate and number of samples used to compute the baseline.

The "monitoring" baseline method is sort of equivalent to the "DigiCam" baseline method if the rate of events is sufficiently high enough.

**Comment by dneise** [comment]
I had a look the different methods, in case you are interested:
https://github.com/dneise/digicam_crab_data_study/blob/master/Compare_digicam_and_pedestal_baseline.ipynb

**Comment by dneise** [comment]
Okay so in this notebook I am starting to compare the hillas parameters for the different baseline strategies.

It takes quite a long time to run and I am not yet done. But in order to perform this study, I think I had to modify digicampipe as well. 

So together with this notebook comes a digigcampipe branch (and a PR) ... so despite this not being ready yet... here is the notebook and a link to the branch.

 * [notebook](https://github.com/dneise/digicam_crab_data_study/blob/master/Compare_hillas_parameters_using_different_baseline_strategies.ipynb)
 * [branch: crab_analysis_study_dneise](https://github.com/cta-sst-1m/digicampipe/tree/crab_analysis_study_dneise)
 * [PR](https://github.com/cta-sst-1m/digicampipe/pull/147)

**Comment by calispac** [comment]
So here is a link to what I was mentioning (html file): 
https://cernbox.cern.ch/index.php/s/WB1L4OyEOXKLMdQ

**Comment by dneise** [comment]
> So here is a link to what I was mentioning (html file):
https://cernbox.cern.ch/index.php/s/WB1L4OyEOXKLMdQ

This looks very interesting thanks.

**Comment by dneise** [comment]
Update of the notebook above ... no real difference visible between the 3 possibilities

**Comment by dneise** [comment]
Oh .. I found my bug ... the results looked too similar. Turned out I had a but resulting not using different analysis at all. :-( ... will take another few hours to execute the correct stuff ... sorry

**Comment by dneise** [comment]
Okay .. so I executed the notebook above ...but just for 3 files as an example:
https://github.com/dneise/digicam_crab_data_study/blob/master/foo/Compare_hillas_parameters_using_different_baseline_strategies-test.ipynb

We see this goes in the right direction ... the results are not totally identical but look similar.

I now have to get on the bus .. so the complete results will only be available some time tonight sorry.

**Comment by dneise** [comment]
Okay .. my notebook above is now finished... I have no idea what to conclude, but we have atleast the plots to look at. Looking forward to your comments or questions.

**Comment by moderski** [comment]
@dneise - you have done an excellent work :-), but I think this kind of "high level" analysis (comparing Hills parameters) is always prone to an accusation of not using the optimal method of image cleaning, charge reconstruction, etc.
I think the best way to compare different baseline strategies would be through comparison of charge resolution from both Monte Carlo and lab measurements with CTS.

**Comment by dneise** [comment]
@moderski you are very diplomatic :-D Others might have said: "That study was a complete and utter waste of our time"... and I think you are right, Rafal. There might not be much to learn from this study. The cleaning argument is indeed very valid. I think the cleaning levels were not optimized for any of the 3 variants, so the results might be quite crappy.

Unfortunately I cannot help with the monte carlos (apart from the toy monte carlo) since I do not know anything about it ... neither where the software is .. nor how to use it and so on and so on. Lab measurements with CTS sound very interesting indeed.... but I guess I cannot go to geneva withing the next days and help with it :-|

**Comment by dneise** [comment]
So .. when does @Etienne12345 need to know the answer regarding this matter?
